sympy/utilities/tests/test_code_quality.py

```diff
@@ -178,6 +180,8 @@ def test_this_file(fname, test_file):
             if (implicit_test_re.search(line) and
                     not filter(lambda ex: ex in fname, import_exclude)):
                 assert False, message_implicit % (fname, idx + 1)
+            if ".func is" in line and "test" not in fname:
```

Comment:
There may be whitespace between `.` and `func` and/or between `func` and `is`. Why don't use regex like other checks? Also use `test_file_re` like line 162 above or add `else` to it and check `.func is` there.

Reply:
Yeah, regex would probably be a good idea. I didn't know about that whitespace case, though. Can there be an arbitrary number of spaces between any of them?

Reply:
Based on limited experience, it seems that the intervening whitespace cannot include a newline so using a regex on line (logical or physical, I'm not sure which is being referenced in the code shown) should work. Would an unlikely, but possibly false positive, occur if the text appeared in a comment or string on the line? e.g. `print("do not use foo.func is bar when testing")`

Reply:
From Python documentation:
> Except at the beginning of a logical line or in string literals, the whitespace characters space, tab and formfeed can be used interchangeably to separate tokens.

Of course, it's bad coding style, but `code_quality` test is for checking it.

Reply:
Interesting. The regex I used is ```\.\s*func\s+is```. Based on some tests, I think that might work. I guess a string or comment false positive could occur, but I'm not sure how to test for that. Is it likely to happen?